src/app/data-explorer/share-data-specification-dialog/share-data-specification-dialog.component.ts

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022-2023 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+import { Component, Inject, OnInit, Optional } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+export interface ShareDataSpecificationDialogOptions {
+  sharedWithCommunity: boolean;
+}
+
+export interface ShareDataSpecificationDialogResponse {
+  sharedWithCommunity: boolean;
+}
+
+@Component({
+  selector: 'mdm-share-data-specification-dialog',
+  templateUrl: './share-data-specification-dialog.component.html',
+  styleUrls: ['./share-data-specification-dialog.component.scss'],
+})
+export class ShareDataSpecificationDialogComponent implements OnInit {
+  initialSharedValue: boolean;
+  dialogCheckboxValue: boolean;
```

Comment:
I think I'm seeing some code duplication in this dialog component:

1. `SharedDataSpecificationDialogOptions` and `SharedDataSpecificationDialogResponse` are exactly the same, so perhaps there should just be one interface type.
2. `initialSharedValue` is set but then is never used, instead `dialogCheckboxValue` is the property it cares about. Why not just use `dialogCheckboxValue` alone?

Reply:
1. Done
2. Well spotted initialSharedValue is now used to disable the "share" button if there are no changes.